src/Raven.Server/Documents/ShardedHandlers/ShardedStatsHandler.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Operations;
+using Raven.Client.Http;
+using Raven.Client.Util;
+using Raven.Server.Documents.ShardedHandlers.ShardedCommands;
+using Raven.Server.Documents.Sharding;
+using Raven.Server.Routing;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.ShardedHandlers
+{
+    public class ShardedStatsHandler : ShardedRequestHandler
+    {
+        [RavenShardedAction("/databases/*/stats", "GET")]
+        public async Task Stats()
+        {
+            var op = new ShardedStatsOperation();
+
+            var statistics = await ShardExecutor.ExecuteParallelForAllAsync(op);
+            statistics.Indexes = GetDatabaseIndexesFromRecord();
+            statistics.CountOfIndexes = statistics.Indexes.Length;
+
+            using (ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            {
+                await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBodyStream()))
+                {
+                    Json.BlittableJsonTextWriterExtensions.WriteDatabaseStatistics(writer, null, statistics);
+                }
+            }
+        }
+
+        private IndexInformation[] GetDatabaseIndexesFromRecord()
+        {
```

Comment:
Wouldn't it be better to move this inside the ShardedStatsOperation? Then each time I execute this (e.g. in a different place than this endpoint) the results will be consistent. cc @karmeli87 Thoughts?